### PR TITLE
M3-4482 Add: Ability to upload a TLS certificate for a Bucket

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Fill in the SSL tab (which may end up renamed to Settings) with a form to allow users to upload an SSL/TLS cert

Includes: 

- JS client methods for uploading, getting, and deleting certs
- BucketSSL component

Todo: 

- [ ] Unit test the things
- [ ] Add a link to the Linode doc below (once it's been updated to reflect this capability).

## Note to Reviewers

There were no designs for this, and I had to make a bunch of judgment calls. I decided to hide the SSL inputs if you already have a cert on your bucket, since trying to submit the form in this case always causes a "Please delete existing certs first" error. Please give feedback on the approach.

To test this properly (successfully upload a cert), you have to create a cert that's valid for the bucket. I followed a combination of these two guides:

https://www.linode.com/docs/platform/object-storage/host-static-site-object-storage/
https://medium.com/@deepak13245/hosting-a-https-static-websites-using-s3-and-lets-encrypt-6f3e53014ff2

Reach out if you run into trouble.

### CODE REVIEW

File: packages/manager/src/features/ObjectStorage/BucketDetail/BucketSSL.tsx [hunk 1]
```diff
@@ -1,7 +1,228 @@
...
+  );
+
+  const hasSSL = request.data;
+
+  if (request.loading) {
+    return <CircleProgress />;
+  }
+
+  if (request.error) {
+    return <ErrorState errorText="Error loading TSL cert data" />;
```
Comment (WilkinsKa1) [inline packages/manager/src/features/ObjectStorage/BucketDetail/BucketSSL.tsx:78]: Still going through testing steps, but noticed this- 'TSL'

**Review by johnwcallahan — COMMENTED** [review]
Looks great so far. I haven't tested it end-to-end yet (just mocking).

Seems like the forms could be closer together? (just a matter of taste... I'll defer to @tiffwong or @WilkinsKa1). Also seems like each form should be full-width on mobile.

The wording `Are you sure you want to remove all certificates from this Bucket?` might be confusing, since if they remove their custom cert, the default one will still be there.

Other than that, UI/UX/functionality looks good. I'll do a full feature test later this morning.

**Review by tiffwong — APPROVED** [review]
After much effort, it worked lol

**Comment by tiffwong** [comment]
> Looks great so far. I haven't tested it end-to-end yet (just mocking).
> 
> Seems like the forms could be closer together? (just a matter of taste... I'll defer to @tiffwong or @WilkinsKa1). Also seems like each form should be full-width on mobile.
> 
> The wording `Are you sure you want to remove all certificates from this Bucket?` might be confusing, since if they remove their custom cert, the default one will still be there.
> 
> Other than that, UI/UX/functionality looks good. I'll do a full feature test later this morning.

I agree, maybe we could increase the width of the text fields and leave a 40px gap or something

**Review by johnwcallahan — APPROVED** [review]
I was able to able to get it working end-to-end; works really well. In addition to the docs link, I think we should include some basic information about cert requirements in the helper text. We could even make it dynamic and use the name of the bucket in the examples.

**Comment by Jskobos** [comment]
Going to merge this so we can demo from develop, but will add additional helper text when we add the doc

**Pull request merged**